File: python/yanghui/yh.py
#!/usr/bin/env python
#encoding:utf-8


def displayData(line):
    arr=[([0]*line)  for i in range(line)]
    arr[0][0]=1

    for i in range(1,line):
        for j in range(i+1):
            if j==0:
                arr[i][0]=1
                continue

            if j==i:
                arr[i][j]=1
            else:
                arr[i][j] = arr[i-1][j-1]+arr[i-1][j]



    for i in range(line):
        for j in range(line-i):
            print " ",
        for j in range(len(arr[i])):
            if arr[i][j]>0:
                print arr[i][j], " ",
        print "\n"

    
    return





if __name__=="__main__":
    line=7
    displayData(line)

    

File: go/yanghui/yh.go
package main

import (
	"fmt"
)

func displayData(line int) {
	data := make([][]int, 7, 7)

	for i := 0; i < line; i++ {
		tmp := make([]int, 7, 7)
		data[i] = tmp
	}

	data[0][0] = 1
	for i := 1; i < line; i++ {
		for j := 0; j <= i; j++ {
			if j == 0 {
				data[i][0] = 1
				continue
			}

			if i == j {
				data[i][j] = 1
			} else {
				data[i][j] = data[i-1][j-1] + data[i-1][j]
			}
		}
	}

	for i := 0; i < line; i++ {
		// 空格
		for j := 0; j < line-i; j++ {
			fmt.Printf("  ")
		}

		// 显示数字
		for _, v := range data[i] {
			if v > 0 {
				fmt.Printf("%2d  ", v)
			}
		}
		fmt.Printf("\n")
	}
}

func main() {
	line := 7
	fmt.Printf("显示%d行\n", line)

	displayData(line)
}
